components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AbstractApplicationAuthenticator.java

```diff
@@ -62,10 +64,13 @@ public AuthenticatorFlowStatus process(HttpServletRequest request,
         // if an authentication flow
         if (!context.isLogoutRequest()) {
             if (!canHandle(request)
-                || (request.getAttribute(FrameworkConstants.REQ_ATTR_HANDLED) != null && ((Boolean) request
-                    .getAttribute(FrameworkConstants.REQ_ATTR_HANDLED)))) {
+                    || (context.getProperty(FrameworkConstants.RETRYING_STEP) != null
+                        && ((Boolean) context.getProperty(FrameworkConstants.RETRYING_STEP)))
```

Comment:
Shall we use "Boolean.TRUE.equals(FrameworkConstants.RETRYING_STEP)"? will avoid the class cast exceptions also.